src/main/java/org/dmfs/oauth2/client/BasicOAuth2AuthorizationRequest.java

```diff
@@ -108,12 +94,8 @@ public URI authorizationUri(URI authorizationEndpoint)
         // TODO: refuse to return a URI without a client id.
         try
         {
-            // this is not optimal, but we need to create two URIs to make sure we don't append our query to an existing query. That's probably the easiest way
-            // for now.
-            return URI.create(String.format("%s?%s",
-                    new URI(authorizationEndpoint.getScheme(), authorizationEndpoint.getAuthority(),
-                            authorizationEndpoint.getPath(), null, null).toASCIIString(),
-                    new XWwwFormUrlEncodedEntity(mParameters).toString()));
+            return URI.create(new URI(authorizationEndpoint.getScheme(), authorizationEndpoint.getAuthority(),
```

Comment:
No, for now I've only switched occurrences of redirect uris. URIs that will be used against http-client-essentials still need to be Java's URI until http-client-essentials uses uri-toolkit too.